compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -1812,6 +1827,20 @@ private BType checkStructFieldAccess(BLangVariableReference varReferExpr, Name f
             return symTable.errType;
         }
 
+        // If it's an index-based access, the return type is a nil-able.
+        if (varReferExpr.getKind() == NodeKind.INDEX_BASED_ACCESS_EXPR) {
+            Set<BType> types = new LinkedHashSet<>();
```

Comment:
I think logic is same as `checkTypeForIndexBasedAccess(...)`.

Reply:
This code was unnecessary. Hence removed it.